sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -4601,6 +4601,231 @@ case class ArrayExcept(left: Expression, right: Expression) extends ArrayBinaryL
     newLeft: Expression, newRight: Expression): ArrayExcept = copy(left = newLeft, right = newRight)
 }
 
+@ExpressionDescription(
+  usage = "_FUNC_(x, pos, val) - Places val into index pos of array x (array indices start at 1)",
```

Comment:
snowflake starts at 0.

Reply:
https://github.com/apache/spark/pull/38867#discussion_r1060395813

Reply:
Basically I don't think we should be exactly reproducing snowflake; there are some clear differences between Spark and Snowflake, including the 1-based indexing. But I'm happy to follow your guidance